pkg/steps/cluster_claim.go

```diff
@@ -180,7 +178,67 @@ func mutate(secret *corev1.Secret, name, namespace string) (*corev1.Secret, erro
 	}, nil
 }
 
-func ClusterClaimStep(as string, clusterClaim *api.ClusterClaim, hiveClient ctrlruntimeclient.Client, client loggingclient.LoggingClient, jobSpec *api.JobSpec) api.Step {
+var (
+	waitForClaim = func(client watchingclient.Client, ns, name string, timeout time.Duration) error {
```

Comment:
I am not sure how to unit testing this.

/hold

Use built image for rehearsals.

Reply:
Why is this in a global?

cc @alvaroaleman for how to unit-test

Reply:
To be precise, I'd like to unit-test the caller `func (s *clusterClaimStep) run(ctx context.Context) error` of this function.
The function is global because I want to override it in the unit tests of the caller to avoid timeout.

Reply:
Make it as step's field. See if it is better.